removefiles/removefiles.go

```diff
@@ -0,0 +1,63 @@
+package removefiles
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"slices"
+	"strings"
+)
+
+// Remove deletes any files or directories in dir (and sub-directories) whose
+// names match one of the values in remove, then returns a count of the items
+// deleted.
+//
+// The remove parameter should be a comma delimited list of file names, e.g.
+// "Thumbs.db,.DS_Store".  If remove is empty, then Remove returns without
+// deleting anything.
+func Remove(dir, remove string) (int, error) {
+	var count int
+
+	if remove == "" {
+		return 0, nil
+	}
+
+	fi, err := os.Stat(dir)
+	if err != nil {
+		return 0, fmt.Errorf("remove: %v", err)
+	}
+	if !fi.IsDir() {
+		return 0, fmt.Errorf("remove: path: %q: not a directory", dir)
+	}
+
+	files := parseFilenames(remove)
+	err = filepath.WalkDir(
+		dir,
+		func(path string, d fs.DirEntry, err error) error {
+			if slices.Contains(files, d.Name()) {
```

Comment:
Should we avoid deleting directories matching the remove filenames?

Reply:
I originally did exclude directories from deletion, but then thought maybe we'd want to delete hidden directories in some cases? Deleting directories is not necessary to meet the requirements of issue #2 which only mentions ".DS_Store" files by name.  Maybe there is a case for deleting a ".git" directory, or something similar?

Other considerations:

1. The a3m "remove_files" script looks like it deletes directories as well as files: https://github.com/artefactual-labs/a3m/blob/main/a3m/client/clientScripts/remove_unneeded_files.py#L32
2. The default list of "removable_files" for a3m ("Thumbs.db, Icon, Icon\r, .DS_Store" ) doesn't include any directories: https://github.com/artefactual-labs/a3m/blob/main/a3m/settings/common.py#L121C19-L121C53
3. Deleting a directory containing multiple files (e.g. "adir/file1, adir/files" )within "WalkDir" could cause unanticipated results. Maybe a file not found error because "adir" is deleted, then it tries to walk the dir and delete "adir/file1"? The count may or may not include the files in "adir".